### PR TITLE
Remove redundant throws

### PR DESCRIPTION
**Pull request by altro3**

### CODE REVIEW

File: data-jdbc/src/main/java/io/micronaut/data/jdbc/operations/DefaultJdbcRepositoryOperations.java [hunk 1]
```diff
@@ -826,11 +826,7 @@ private ConnectionContext getConnectionCtx() {
     @Override
     public <R> R execute(@NonNull ConnectionCallback<R> callback) {
         return executeWrite(connection -> {
-            try {
-                return callback.call(connection);
-            } catch (SQLException e) {
-                throw new DataAccessException("Error executing SQL Callback: " + e.getMessage(), e);
```
Comment (sdelamo) [inline data-jdbc/src/main/java/io/micronaut/data/jdbc/operations/DefaultJdbcRepositoryOperations.java:832]: Please, revert this change. This will be breaking change if someone relies on catching a `DataAccessException`.

Comment (altro3) [reply data-jdbc/src/main/java/io/micronaut/data/jdbc/operations/DefaultJdbcRepositoryOperations.java:832]: Yes, okay, returned it, but the problem is that this exception will never happen

изображение

Need to add `throws SQLException` to interface method

**Review by sdelamo — APPROVED** [review]
